larray/core.py

```diff
@@ -8781,8 +8798,13 @@ def split_axis(self, axis, sep='_', names=None):
         a\\b | b0 | b1 | b2
          a0 |  0 |  1 |  2
          a1 |  3 |  4 |  5
+        >>> regex = '([a-z0-9]+).([a-z0-9]+)'
```

Comment:
I would have hoped for a more interesting example (which cannot work using the usual sep argument). And if using "combined" as starting point, I would use "(\w+)_(\w+)" as the regex (and avoid defining it in a separate variable).
An example I would find more interesting is the same labels than combined but without the delimiter:
a0b0, a0b1 etc (assuming < 9 labels in each dim ;-)).
```python
new = combined.set_labels(x.a_b, [l.replace('_', '') for k in combined.a_b.labels])
```
or (maybe clearer) set the labels explicitly:
```python
new = combined.set_labels(x.a_b, ['a0b0', 'a0b1', 'a0b2', 'a1b0', 'a1b1', 'a1b2'])
```
PS: we should allow this syntax in set_labels (to be consistent with array creation funcs):
```python
new = combined.set_labels(x.a_b, 'a0b0,a0b1,a0b2,a1b0,a1b1,a1b2')
```
I'll open another issue for this.